dali/sasha/saarch.cpp

```diff
@@ -85,295 +85,444 @@ static void mkDateCompare(bool dfu,const char *dt,StringBuffer &out,char fill)
-                                name.append(',').append(val);
-                            tok = strtok_r(NULL, "|,",&saveptr);
-                        }
-                        free(parse);
+
+                    if (sortInc)
+                    {//The output should contain WUs from new to old. So, we need to reverse the order before output.
+                        Owned<CWUData> wu = new CWUData();
+                        wu->wuTree.setown(wuTree.getClear());
+                        if (outputModifiedTime)
+                            fileIterator->getModifiedTime(wu->modifiedTime);
+                        wus.append(*wu.getClear());
                     }
-                    cmd->addId(name.str());
-                    if (cmd->getAction()==SCA_GET) {
-                        StringBuffer xml;
-                        toXML(&pt,xml);
-                        if (!cmd->addResult(xml.str()))
+                    else
+                    {
+                        CDateTime dt;
+                        if (outputModifiedTime)
+                            fileIterator->getModifiedTime(dt);
+
+                        if (!addOutput(outputFields, outputModifiedTime, hasWUSOutput, wuTree, dt, WUSbuf))
+                        {
+                            overflowed = true;
                             break;
+                        }
                     }
+                    wuCount++;
+                    if (wuCount == numWUs)
+                        break;
                 }
-                if (index>start+num)
+                if (overflowed || (wuCount == numWUs))
                     break;
             }
+            if (sortInc)
+            {
+                ForEachItemInRev(i, wus)
+                {
+                    CWUData &wuData = wus.item(i);
+                    if (!addOutput(outputFields, outputModifiedTime, hasWUSOutput, wuData.wuTree, wuData.modifiedTime, WUSbuf))  //Log an error?
+                        break;
+                }
+            }
+
+            if (hasWUSOutput)
+                cmd->setWUSresult(WUSbuf);
         }
-    }
-    if (haswusoutput)
-        cmd->setWUSresult(WUSbuf);
+        void getOnlineWUs()
+        { //The code here is refactered based on the pre-existing code.
+            if (cmd->getAction()==SCA_WORKUNIT_SERVICES_GET)
+                throw MakeStringException(-1,"SCA_WORKUNIT_SERVICES_GET not implemented for online workunits!");
+
+            Owned<IRemoteConnection> conn;
+            setSDSConnection(conn, mask);
+            if (!conn)
+                return;
+    
+            CIArrayOf<CWUData> wus;
+            unsigned wuCount = 0;
+            StringBuffer wuid;
+            Owned<IPropertyTreeIterator> iter = conn->queryRoot()->getElements(isWild ? "*" : NULL);
+            ForEach(*iter)
+            {
+                IPropertyTree *pt = &iter->query();
+                wuid.set(pt->queryName());
+                if (wuid.isEmpty())
```

Comment:
Should be impossible, so curious why this check is here?

Reply:
This check will be removed.